mlir/lib/Dialect/Linalg/IR/LinalgOps.cpp

```diff
@@ -806,6 +806,36 @@ struct FoldFillWithTensorExtract : public OpRewritePattern<tensor::ExtractOp> {
   }
 };
 
+/// Fold tensor.extract_slice(linalg.fill(<input>)) into <input>
```

Comment:
Hm, this comment is not quite correct, is it? You are replacing `tensor.extract_slice` with `linalg.fill`, right?

What if there are multiple uses of the original `linalg.fill`? Then this is adding another `linalg.fill` (and `tensor.empty`) rather than updating the original `linalg.fill`? Is that still more canonical?

Reply:
I think even with multiple users it might be ok to introduce a new fill of the smaller shape. Thats an opinion though. I think all such slice propagation patterns need to be kept separate, but `fill -> slice` might be worth adding to canonicalization.